errors.go

```diff
@@ -98,16 +98,23 @@ func (ve *ValidationError) Error() string {
 	return fmt.Sprintf("jsonschema: %s does not validate with %s: %s", quote(leaf.InstanceLocation), u+"#"+leaf.KeywordLocation, leaf.Message)
 }
 
-func (ve *ValidationError) GoString() string {
+func (ve *ValidationError) goStringTo(tgt *strings.Builder, indent int) {
 	sloc := ve.AbsoluteKeywordLocation
 	sloc = sloc[strings.IndexByte(sloc, '#')+1:]
-	msg := fmt.Sprintf("[I#%s] [S#%s] %s", ve.InstanceLocation, sloc, ve.Message)
+	for i := 0; i < indent; i++ {
+		tgt.WriteByte(' ')
+	}
+	tgt.WriteString(fmt.Sprintf("[I#%s] [S#%s] %s", ve.InstanceLocation, sloc, ve.Message))
 	for _, c := range ve.Causes {
-		for _, line := range strings.Split(c.GoString(), "\n") {
-			msg += "\n  " + line
-		}
+		tgt.WriteByte('\n')
+		c.goStringTo(tgt, indent+2)
```

Comment:
why `indent+2`? should this be `indent+1`

Reply:
`indent` value actually for space number .
the original msg using `\n + space + space`.
if write space for loop using `indent * 2`, `indent` could `+1`

Reply:
Here are three choices:
1. for loop write space bytes use `indent * 2` and `goStringTo` `indent+1` keep original output style
2. make output more compact with one space offset new line
3. keep these codes not change